### DISCUSSION TITLE
How to validate entity versus current state?

### DISCUSSION THREAD

**Discussion by leandrodotec** [Avanade/Beef#181] [Q&A] [answered]
Hi,
We have a situation where we need to validate the posted value against the value that is saved in the database. 

Do we have a validator for that? When validating one parameter the code doesn't look that bad, but the problem happens when we have entities with multiple parameters that can be changes, and others that are read only.

We did a custom validator like this:

```
public class ClaimValidator : Validator<Claim>
    {
       private readonly IClaimManager _claimManager
        public ClaimValidator (IClaimManager claimManager)
        {
            _claimManager = claimManager
            Property(x => x.DocumentId).Mandatory();
            Property(x => x.PatientName)
                    .Custom(PatientNameCantChangeWhenReadOnly);
        }
        private async Task PatientNameCantChangeWhenReadOnly(PropertyContext<Claim, string> context)
        {
            var current = _claimManager.GetAsync(context.Parent.Value.Id);
            if(current.ReadyOnly && !current.PatientName.Equals(context.Value)) 
            {
                context.CreateErrorMessage("Ca't change patientName when is ready only.");
            }
        }
    }
```
In advance, thanks for the help.

**Comment by chullybun** [comment, marked as answer]
Hi @leandrodotec, first up apologies for delayed response. There is currently no out-of-the-box validator to verify whether certain fields are read-only and check they have not changed. The way you have it is the long-hand way I would do for a single property. If you had more than one property then I would override the `OnValidateAsync` method and do there as more efficient.

The validation framework is extensible, so you could add a new if needed, even contribute it back into beef :-)

**Reply by leandrodotec** [reply]
Thanks Eric. We do have other situations with multiple fields, so we will use OnValidateAsync.
